src/randoop/main/GenTests.java

```diff
@@ -429,264 +399,335 @@ public boolean handle(String[] args) throws RandoopTextuiException {
-          if (GenInputsAbstract.test_classes.matcher (v.getType().getName())
-              .matches()) {
-            keep = true;
-            break;
-          }
+    
+    if (! GenInputsAbstract.no_regression_tests) {
+      List<ExecutableSequence> regressionSequences = explorer.getRegressionSequences();
+      if (regressionSequences.size() > 0) {
+        if (! GenInputsAbstract.noprogressdisplay) {
+          System.out.format("%nRegression test output:%n");
+        }
+        outputTests(regressionSequences, GenInputsAbstract.regression_test_filename);
+      } else {
+        if (! GenInputsAbstract.noprogressdisplay) {
+          System.out.format("No regression tests to output%n");
         }
-        if (keep)
-          tc_seqs.add (es);
       }
-      sequences = tc_seqs;
-      System.out.printf ("%n%d sequences include %s%n", sequences.size(),
-                         GenInputsAbstract.test_classes);
     }
+    
+    return true;
+  }
 
-    // If specified remove any sequences that are used as inputs in other
-    // tests.  These sequences are redundant.
-    //
-    // While we're at it, remove the useless sequence "new Object()".
-    Sequence newObj = new Sequence().extend(objectConstructor);
-    if (GenInputsAbstract.remove_subsequences) {
-      List<ExecutableSequence> unique_seqs
-        = new ArrayList<ExecutableSequence>();
-      Set<Sequence> subsumed_seqs = explorer.subsumed_sequences();
-      for (ExecutableSequence es : sequences) {
-        if (!subsumed_seqs.contains(es.sequence) && !es.sequence.equals(newObj)) {
-          unique_seqs.add(es);
-        }
+  /**
+   * Builds the test predicate that determines whether a particular sequence will
+   * be included in the output based on command-line arguments.
+   * 
+   * @param objectConstructor  the constructor for the Object class
+   * @return the predicate 
+   */
+  private Predicate<ExecutableSequence> createTestOutputPredicate(ConstructorCall objectConstructor) {
+    Predicate<ExecutableSequence> outputTest;
+    if (GenInputsAbstract.dont_output_tests) {
+      outputTest = new AlwaysFalse<>();      
+    } else {
+      Predicate<ExecutableSequence> baseTest;
+      //base case: exclude sequences with just "new Object()", keep everything else
+      //to exclude something else, add sequence to excludeSet
+      Sequence newObj = new Sequence().extend(objectConstructor);
+      Set<Sequence> excludeSet = new LinkedHashSet<>();
+      excludeSet.add(newObj);
+      baseTest = new ExcludeTestPredicate(excludeSet);
+      if (GenInputsAbstract.include_only_classes != null) { //keep only tests with test classes
+        baseTest = baseTest.and(new IncludeTestPredicate(GenInputsAbstract.include_only_classes));
       }
-      if (!GenInputsAbstract.noprogressdisplay) {
-        System.out.printf("%d subsumed tests removed%n", sequences.size() - unique_seqs.size());
+
+      //Use arguments to determine which kinds of tests to output
```

Comment:
I had assumed that test generation and test outputting would be distinct -- that is, the same tests would be generated and put into two lists, regardless of the outputting parameters.  Then the no_error_revealing_tests and no_regression_tests flags would be used when deciding whether to output the tests.  That would simplify the logic during generation.  (Might it help to ensure determinism too?  I'm not sure.)  But maybe this approach is better for some reason such as efficiency?